### DISCUSSION TITLE
Where do I find "raise notice" messages?

### DISCUSSION THREAD

**Discussion by enyo** [supabase/supabase#9385] [Questions] [answered]
I'm using `raise notice 'No accessories are available to gift.';` in a trigger function in my postgresql database, but I can't find it anywhere (maybe it's due to a bug in my code, but I can't find it).

Should these messages appear in the "Postgres logs" view in the dashboard?

<img width="218" alt="image" src="https://user-images.githubusercontent.com/133277/194518645-1e325bde-0488-4cb0-9ee2-97310d9bfc14.png">

**Comment by GaryAustin1** [comment, marked as answer]
Use raise log instead.  That goes to the Postgres log.

**Reply by tvogel** [reply]
And how can I output from `plpgsql` in the query editor?
```
do $$
begin
  raise notice 'Hello world!';
end;
$$;
```
succeeds but shows nothing. I could `raise exception` but that does not work for general output...

**Reply by enyo** [reply]
Maybe I’m missing something @tvogel, but as @GaryAustin1 said you should use `raise log` instead.

**Reply by tvogel** [reply]
@enyo That's a fine workaround but for fiddling around, I think, it would be great to have a different channel that is not necessarily logged but instead put into the query editors output pane. It would also be much less cumbersome than switching tabs and refreshing the log view on each turnaround.

**Reply by devhandler** [reply]
@GaryAustin1 any reason why we cannot use raise notice?

using raise log and then looking at supabase log ui is ok but not convenient. when i use pg admin, the raise notice msg will be passed to the client(pg admin), but wondering why raise notice msg cannot be passed to a js client (say, react, vue, etc.)

**Reply by GaryAustin1** [reply]
I can't answer what could be done,  but the clients are not connected to the database thru the database port.  There is no feedback to a client.  You can only do REST API requests which connect and go away.  So logs would have to be stored (like they are now) and then read from a client.  I guess there could be some sort of realtime method done, but not sure what kind of priority that would get.

**Reply by Ziinc** [reply]
To update the answer for the main OP, pg_audit is now the advised method for configuring postgres logging levels. You can configure the log levels to `INFO` or `DEBUG1` if you so desire. Please refer to [official logging docs here](https://supabase.com/docs/guides/platform/logs#configuring-pgauditlog)

I highly recommend adjusting the log levels only when debugging, otherwise it would get very noisy.

By default, query logs are turned off to prevent exposing sensitive information accidentally.

--- 

To answer @devhandler : 
Error messages via `RAISE` will be returned to clients and can be customized, as per the [postgrest docs](https://postgrest.org/en/latest/references/errors.html#raise-errors-with-http-status-codes).

Using `RAISE NOTICE` to return information to the client is an antipattern and would be abusing the logging system to perform payload responses or data streams. Furthermore, `RAISE NOTICE` and other non-error log levels would not abort the transaction (as per the [postgres docs](https://www.postgresql.org/docs/current/plpgsql-errors-and-messages.html)).

If you wish to return specific information or resource to the client, you should return it in the function result for the api call.

**Reply by Ziinc** [reply]
Thanks @tvogel for the ui feedback, it would be technically feasible to achieve the console logging experience. I'll put some thought into this feedback.

**Reply by roo12312** [reply]
raise log doesn't work I'm trying to log this on function run 
`raise log 'starting';`
I'm not finding it in the logs section

**Comment by EskelCz** [comment]
After an hour of tinkering I found out that it has to be uppercase, for some reason. `RAISE LOG`
Infuriating.

**Reply by karmasakshi** [reply]
Works in lowercase too.

**Comment by Radiokot** [comment]
Hey,
In case someone stumbles upon this, you can use `RAISE WARNING` which stands out among the regular blue log records.

**Comment by Jasper-Nelligan** [comment]
I'm still unable to see any logs triggered in my RPC functions. I've tried RAISE NOTICE, RAISE LOG, RAISE WARNING... nothing. I'm running Supabase studio locally on my computer. I've also tried enabling `pgaudit` but that didn't do anything. Any suggestions on what to try next?
